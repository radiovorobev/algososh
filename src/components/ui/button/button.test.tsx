import renderer from "react-test-renderer";
import {Button} from "./button";
import {screen, fireEvent, render} from "@testing-library/react";

describe("Button test OK", () => {

    it('Button with text OK', () => {
        const tree = renderer.create(<Button text='Текст кнопки' />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Button w/o text OK', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('disabled Button OK', () => {
        const tree = renderer.create(<Button disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('loading Button OK', () => {
        const tree = renderer.create(<Button isLoader={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('clicked Button OK', () => {
        const mockOnClick = jest.fn();
        render(<Button text='Текст кнопки' onClick={mockOnClick} />);
        const button = screen.getByText('Текст кнопки');
        fireEvent.click(button);
        expect(mockOnClick).toHaveBeenCalled();
    });
});