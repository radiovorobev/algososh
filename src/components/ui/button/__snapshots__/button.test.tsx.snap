// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button w/o text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст кнопки
  </p>
</button>
`;

exports[`Button button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст кнопки
  </p>
</button>
`;

exports[`Button disabled Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button loading Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test OK Button w/o text OK 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button test OK Button with text OK 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст кнопки
  </p>
</button>
`;

exports[`Button test OK disabled Button OK 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button test OK loading Button OK 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button tests button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст кнопки
  </p>
</button>
`;
